src/sources/eventstoredb/mod.rs

```diff
@@ -0,0 +1,129 @@
+use self::types::{
+    EventStoreDbMetricsHttpError, EventStoreDbMetricsReceived, EventStoreDbStatsParsingError, Stats,
+};
+use crate::{
+    config::{self, GlobalOptions, SourceConfig, SourceDescription},
+    shutdown::ShutdownSignal,
+    Event, Pipeline,
+};
+use futures::{stream, FutureExt, SinkExt, StreamExt};
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+pub mod types;
+
+#[derive(Deserialize, Serialize, Clone, Debug, Default)]
+struct EventStoreDbConfig {
+    #[serde(default = "default_endpoint")]
+    endpoint: String,
+    #[serde(default = "default_scrape_interval_secs")]
+    scrape_interval_secs: u64,
+    default_namespace: Option<String>,
+}
+
+pub fn default_scrape_interval_secs() -> u64 {
+    3
+}
+
+pub fn default_endpoint() -> String {
+    "https://localhost:2113/".to_string()
+}
+
+inventory::submit! {
+    SourceDescription::new::<EventStoreDbConfig>("eventstoredb_metrics")
+}
+
+impl_generate_config_from_default!(EventStoreDbConfig);
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "eventstoredb_metrics")]
+impl SourceConfig for EventStoreDbConfig {
+    async fn build(
+        &self,
+        _name: &str,
+        _globals: &GlobalOptions,
+        shutdown: ShutdownSignal,
+        out: Pipeline,
+    ) -> crate::Result<super::Source> {
+        eventstoredb(
+            self.endpoint.as_str(),
+            self.scrape_interval_secs,
+            shutdown,
+            out,
+            self.default_namespace.clone(),
+        )
+    }
+
+    fn output_type(&self) -> config::DataType {
+        config::DataType::Metric
+    }
+
+    fn source_type(&self) -> &'static str {
+        "eventstoredb"
+    }
+}
+
+fn eventstoredb(
+    endpoint: &str,
+    interval: u64,
+    shutdown: ShutdownSignal,
+    out: Pipeline,
+    namespace: Option<String>,
+) -> crate::Result<super::Source> {
+    let mut out = out.sink_map_err(|error| error!(message = "Error sending metric: {:?}.", %error));
```

Comment:
```suggestion
    let mut out = out.sink_map_err(|error| error!(message = "Error sending metric.", %error));
```

If this fails to compile, try using `?error` instead of `%error`.

Reply:
done